Sources/SwiftBasicFormat/BasicFormat.swift

```diff
@@ -197,7 +201,25 @@ open class BasicFormat: SyntaxRewriter {
         // don't add a leading newline to the file.
         return true
       }
-      return previousToken.trailingTrivia.endsWithNewline
+      if previousToken.trailingTrivia.pieces.last?.isNewline ?? false {
```

Comment:
Why not `if previousToken.trailingTrivia.endsWithNewline`? Just matching up with `stringSegment`? If so, I don't think anything else uses `endsWithNewline` so you may as well delete it.

Reply:
Because I missed it while changing things to `endsWithWhitespace`. Probably through a series of a rebases or something of the sort. Changed it now.